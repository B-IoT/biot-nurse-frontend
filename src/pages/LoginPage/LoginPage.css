.center-login {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100vh;
}

.login-logo {
  width: 100px;
  margin-bottom: 25px;
}

.login-error {
  margin-top: 25px;
}
