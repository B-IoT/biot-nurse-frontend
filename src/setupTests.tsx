// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter } from 'react-router-dom';
import React, { ComponentType } from 'react';

const queryClient = new QueryClient();
export const router = (children: any) => {
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>{children}</BrowserRouter>
    </QueryClientProvider>
  );
};

const Wrapper: React.ComponentType = ({ children }) => {
  return router(children);
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper as ComponentType, ...options });

global.URL.createObjectURL = jest.fn();

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
