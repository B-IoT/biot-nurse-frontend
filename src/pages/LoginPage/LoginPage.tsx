import { useState } from 'react';

import './LoginPage.css';
import Input from '../../components/input/Input/Input';
import logo from '../../img/logoColor.png';
import Button from '../../components/button/Button/Button';
import { authenticate } from '../../api/api';
import { useHistory } from 'react-router-dom';
import { SEARCH_PATH } from '../../App';
import { translate } from '../../i18n';

const strings = {
  username: translate("username"),
  password: translate("password"),
  wrongCredentials: translate("wrongCredentials"),
  login: translate("login")
}

/**
 * The login page.
 */
export default function LoginPage() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [showError, setShowError] = useState(false);
  const history = useHistory();

  async function handleSignIn() {
    const success = await authenticate(username, password);

    if (success) {
      history.push(SEARCH_PATH);
    } else {
      setShowError(true);
      setTimeout(() => setShowError(false), 3000)
    }
  }

  return (
    <div className="center-login">
      <a href="https://www.b-iot.ch">
        <img className="login-logo" src={logo} alt="BIoT logo" />
      </a>
      <Input
        setKeyword={setUsername}
        defaultText={strings.username!}
        width={350}
        style={{}}
        isPassword={false}
        enterHandler={() => null}
      />
      <Input
        setKeyword={setPassword}
        defaultText={strings.password!}
        width={350}
        style={{ marginTop: 15 }}
        isPassword={true}
        enterHandler={handleSignIn}
      />
      {showError && (
        <div className="error-text login-error" data-testid="error-text">
          {strings.wrongCredentials}
        </div>
      )}
      <Button
        text={strings.login!}
        onClick={handleSignIn}
        width={200}
        style={{ marginTop: 35 }}
      />
    </div>
  );
}
