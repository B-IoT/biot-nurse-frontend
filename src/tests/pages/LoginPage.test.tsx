import { render, router } from '../../setupTests';
import LoginPage from '../../pages/LoginPage/LoginPage';
import renderer from 'react-test-renderer';

it('Login page renders without crashing', () => {
  render(<LoginPage />);
});

it('Login page matches snapshot', () => {
  const rendered = renderer.create(router(<LoginPage />));
  expect(rendered).toMatchSnapshot();
});
