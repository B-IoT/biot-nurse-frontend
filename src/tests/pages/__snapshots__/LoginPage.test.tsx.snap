// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page matches snapshot 1`] = `
<div
  className="center-login"
>
  <a
    href="https://www.b-iot.ch"
  >
    <img
      alt="BIoT logo"
      className="login-logo"
      src="logoColor.png"
    />
  </a>
  <div
    className="search-bar"
    style={
      Object {
        "width": 350,
      }
    }
  >
    <div
      className="search-mask"
      style={
        Object {
          "width": 350,
        }
      }
    >
      <div
        className="search-shadow"
        style={
          Object {
            "width": 450,
          }
        }
      />
      <div
        className="search-light"
        style={
          Object {
            "width": 450,
          }
        }
      />
    </div>
    <div
      className="search-blur"
      style={
        Object {
          "width": 355,
        }
      }
    >
      <input
        className="search-text axiforma-light-blue-21px"
        data-testid="input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Username"
        style={
          Object {
            "width": 275,
          }
        }
        type=""
      />
    </div>
  </div>
  <div
    className="search-bar"
    style={
      Object {
        "marginTop": 15,
        "width": 350,
      }
    }
  >
    <div
      className="search-mask"
      style={
        Object {
          "width": 350,
        }
      }
    >
      <div
        className="search-shadow"
        style={
          Object {
            "width": 450,
          }
        }
      />
      <div
        className="search-light"
        style={
          Object {
            "width": 450,
          }
        }
      />
    </div>
    <div
      className="search-blur"
      style={
        Object {
          "width": 355,
        }
      }
    >
      <input
        className="search-text axiforma-light-blue-21px"
        data-testid="input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Password"
        style={
          Object {
            "width": 275,
          }
        }
        type="password"
      />
    </div>
  </div>
  <div
    className="button css-1i20ack-Hover e128ygwg0"
    onClick={[Function]}
    style={
      Object {
        "marginTop": 35,
        "width": 200,
      }
    }
  >
    <div
      className="button-pressed css-1gvt6rq-FadeIn e128ygwg3"
      style={
        Object {
          "width": 200,
        }
      }
    />
    <div
      className="button-shadow css-1ypgtc4-FadeOut e128ygwg2"
      style={
        Object {
          "width": 200,
        }
      }
    />
    <div
      className="button-light css-1ypgtc4-FadeOut e128ygwg2"
      style={
        Object {
          "width": 200,
        }
      }
    />
    <div
      className="button-text axiforma-medium-blue-22px css-zcw12y-Scale e128ygwg1"
      style={
        Object {
          "width": 200,
        }
      }
    >
      Login
    </div>
  </div>
</div>
`;
